prime-router/src/main/kotlin/azure/BlobAccess.kt

```diff
@@ -105,10 +107,18 @@ class BlobAccess() : Logging {
          * @return the URL for the storage and container that this object represents
          */
         fun getBlobEndpoint(): String {
-            val match = blobEndpointRegex.find(connectionString)
-            val blobStorageUrl = match?.groups?.get("blobEndpoint")?.value
-                ?: throw RuntimeException("Connection string is misconfigured and does not contain a blob endpoint URL")
-            return "$blobStorageUrl/$containerName"
+            val blobEndpointMatch = blobEndpointRegex.find(connectionString)
```

Comment:
Maybe I'm just a regex hater but I feel like it would simpler to just parse the string into a map.

Something simple like this
```kotlin
val map = string
  .split(";")
  .map { it.split("=") }
  .associate { it.first() to it.last() }

val blobEndpoint = map["BlobEndpoint"]
// etc
```